the-collab-lab/website-og
# Optimize (some) images

## Summary

1. I put a couple of our images through [Squoosh](https://squoosh.app/) to see if we could make them a little smaller in size while keeping the same dimensions, without sacrificing apparent quality. The new images have a width suffix in their filename (ex: `two-year-anniversary-1920w.jpg`) to distinguish them from the original assets. There are `jpg` and `webp` versions of each. Switching to these assets brings a data savings of approximately 1.5MB 🎉  
2. I used the HTML `picture` element to serve these new images to the user.

## More on the `picture` element

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture):
> The `<picture>` HTML element contains zero or more `<source>` elements and one `img` element to offer alternative versions of an image for different display/device scenarios.
>
> The browser will consider each child <source> element and choose the best match among them. If no matches are found—or the browser doesn't support the <picture> element—the URL of the `<img>` element's `src` attribute is selected. The selected image is then presented in the space occupied by the <img> element.

Our `source` elements have two attributes:
- `srcset`, which points to the assets that the picture element might render. Each source path is followed by a space, then the intrinsic size of the image asset (in this case, `1920w`).
- `type`, which indicates the type of the data in the source

**⚠️ Note:** The `img` element's `alt` attribute will be applied to the asset selected by the browser; it cannot be applied individually to each `source`.

Here's a code sample from this PR:

``` html
<picture>
  <source
    srcset="assets/img/two-year-anniversart-1920w.webp 1920w"
    type="image/webp">
  <source
    srcset="assets/img/two-year-anniversary-1920w.jpg 1920w"
    type="image/jpg">
  <img
    alt="Celebrating Two Years of Collaboration at The Collab Lab"
    height="1080"
    width="1920"
    src="assets/img/two-year-anniversary-1920w.jpg"
  />
</picture>
```

## Test plan

1. Open [the Deploy Prveview](https://deploy-preview-200--the-collab-lab.netlify.app/)
2. Verify that the two year anniversary hero image looks good
3. Navigate to `About`
4. Verify that the weekly call image looks good.

## Follow-up actions
I think we should 
- audit more of our existing assets to see if they can be optimized, and
- document a process for optimizing images in this repo, which future contributors can references
